From 680f225595b1789bade6ea580f3a728b4650815a Mon Sep 17 00:00:00 2001
From: Rahul T R <r-ravikumar@ti.com>
Date: Tue, 3 Oct 2023 18:20:15 +0530
Subject: [PATCH] v4l2object: Make number of buffers in own pool same as other
 pool

This will set the number of buffers in own pool same as
other pool (From which buffers gets imported). This will improve
the latency and tolerance to back pressure in pipeline, since it
result in enqueuing of all buffers available in the other pool

Signed-off-by: Rahul T R <r-ravikumar@ti.com>
---
 sys/v4l2/gstv4l2object.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index f6cbb44..945a3eb 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -5190,6 +5190,14 @@ gst_v4l2_object_decide_allocation (GstV4l2Object * obj, GstQuery * query)
             min += own_min;
         }
     }
+
+    /* This will set the number of buffers in own pool same as
+     * other pool (From which buffers gets imported). This will improve
+     * the latency and tolerance to back pressure in pipeline, since it
+     * result in enqueuing of all buffers available in the other pool */
+    if (0 != strcmp(obj->vcap.driver, "wave5-dec")) {
+        own_min = min - 1;
+    }
   }
 
   /* Request a bigger max, if one was suggested but it's too small */
-- 
2.34.1

